### ISSUE TITLE
generate_account_sas only generates a blob sas

### ISSUE THREAD

**Issue by AidanStrong** [Azure/azure-sdk-for-python#33177]
- **Package Name**: azure-storage-blob        
- **Package Version**: 12.19.0
- **Operating System**: Ubuntu
- **Python Version**: 3.10.12

**Describe the bug**
I am using the generate_account_sas function to attempt to create a SAS token that is valid for both file and blob services. 

For some reason, this method is part of the blob package. 

**To Reproduce**
```

`   storage_client = StorageManagementClient(CREDENTIAL, SUBSCRIPTION_ID)
    storage_keys = storage_client.storage_accounts.list_keys(resource_group_name, storage_account_full_name)
    storage_keys  =  {v.key_name:  v.value for  v  in  storage_keys.keys}

    # Create an account SAS that's valid for one day
    start_time = datetime.datetime.now(datetime.timezone.utc)
    expiry_time = start_time + datetime.timedelta(days=1)

    # Define the SAS token permissions
    sas_permissions=AccountSasPermissions(read=True, list=True, write=True)

    # Define the SAS token resource types
    # For this example, we grant access to service-level APIs
    sas_resource_types=ResourceTypes(service=True, container=True, object=True)
    print(sas_resource_types)
    sas_token = generate_account_sas(
        account_name=storage_account_full_name,
        account_key=storage_keys['key1'],
        resource_types=sas_resource_types,
        permission=sas_permissions,
        expiry=expiry_time,
        start=start_time
    )

    sas_token = ("SharedAccessSignature=" + sas_token)
    print(sas_token)
    blob_endpoint = "BlobEndpoint=https://{0}.file.core.windows.net/;".format(storage_account_full_name)
    file_endpoint = "FileEndpoint=https://{0}.blob.core.windows.net/;".format(storage_account_full_name)
    connection_string = blob_endpoint + file_endpoint + sas_token
    print(connection_string)
`


```
**Expected behavior**
I'd expect the SAS token that is generated from this to have "ss=fb", but it only has "ss=b", and as a reslt it is only valid for blobs.

**Comment by l0lawrence** [comment]
Hi @AidanStrong thanks for the feedback, we will get back to you asap.

**Comment by jalauzon-msft** [comment]
Hi @AidanStrong, thanks for reaching out. The behavior you describe is just how the SDK is designed today. Each package has its own `generate_account_sas` function that will generate an account SAS with the `ss` parameter hardcoded to that particular service. Unlike our other languages, Python does not have a common package where a common version of this could live.

It would be recommended that you use a separate SAS/connection string for each service that you wish to talk with.

**Comment by vincenttran-msft** [comment]
Hi @AidanStrong although I know it may not be possible for your workflow, it is possible to generate a SAS token compatible with multiple services using Azure Portal. On the left hand side there is a tab for "Shared access signature", and there you can specify the allowed services for the SAS token generated.
<img width="191" alt="image" src="https://github.com/Azure/azure-sdk-for-python/assets/101599632/9a3d179e-b325-4ef8-bef1-2575419efc2b">

However, please do let me know if you have any other questions about the Azure portal, or if this is not workable solution for you.

Thanks!

**Comment by AidanStrong** [comment]
Hi @vincenttran-msft , thanks for getting back. 

Unfortunately this doesn't fit our workflow, but this is exactly what I need. I would have thought it would be possible to create this in python?

**Comment by jalauzon-msft** [comment]
Hi @AidanStrong, unfortunately we do not currently have a way in Python to generate an account SAS that works for both Blob and Files at the same time (without importing from private modules). However, this is something we may consider adding. We will discuss internally about getting this feature added in an upcoming release.

Generating the SAS yourself is possible yet the SAS generation algorithm is a bit complex (as you've likely seen). You can try and look at our internal implementation to see if you can figure out what might be going wrong in your implementation but its a bit abstract and hard to follow directly:
https://github.com/Azure/azure-sdk-for-python/blob/723d47c82bed1ffc609a0952f149c2fe319ac782/sdk/storage/azure-storage-blob/azure/storage/blob/_shared/shared_access_signature.py#L109

From a quick glance at your code though, you might just be missing `encryption_scope` in your calculation (you can use empty string if you do not use an encryption scope). See the [second example in this section](https://learn.microsoft.com/en-us/rest/api/storageservices/create-account-sas#construct-the-signature-string) for API version 2020-12-06 and later.

One other thing to note though is that the stringToSign calculation can change with new API versions and each time we release an SDK version, we automatically use the latest API version available. This means, down the line, if you upgrade your SDK version, and therefore your API version, the SAS calculation may stop working. This does not happen often and there are ways to workaround it, just want to be sure you are aware.
